### DISCUSSION TITLE
How to setup IGDB integration?

### DISCUSSION THREAD

**Discussion by aggband** [tkashkin/GameHub#650] [Q&A] [answered]
First off, I must thank everyone here for their hard work. Apart from this one feature, everything I have attempted so far has worked flawlessly and the end result is a wonderful app which I can't thank all of you enough for. 
Now for the problem. Believe it or not, I think I have have successfully generated an authentication token from IGDB. After following their steps to obtain one, I received said token through the Client credentials grant flow request via POST. However, putting said key in the **"API Key"** section of GameHub results in the following message from https://api.igdb.com/admin/applications:
{
  "message": "Authorization Failure. Have you tried:",
  "Tip 1":   "Ensure you are sending Authorization and Client-ID as headers.",
  "Tip 2":   "Ensure Authorization value starts with 'Bearer ', including the space",
  "Tip 3":   "Ensure Authorization value ends with the App Access Token you generated, NOT your Client Secret.",
  "Docs":    "https://api-docs.igdb.com/#authentication",
  "Discord": "https://discord.gg/FrvfwQg"
}
I have tried with and without the 'Bearer' value in front of API key section, but to no avail. It feels like I am missing something obvious, but no researching so far has yielded results so I thought it would be better to find the answer here instead of some more troubleshooting so that other novices can fix this quickly. I am happy with posting any other information you need.
Thanks for everything.

**Comment by tkashkin** [comment, marked as answer]
Some time ago IGDB merged with Twitch, then they moved their API to Twitch and deprecated the old API. It's unlikely that the old endpoints still work even with new keys.

GameHub wasn't updated to use the new API because it required switching to Twitch API tokens, which required to either
* host a caching proxy server somewhere,
* ask each user to register their own Twitch apps and set their own keys.

Both options are much less practical than the old API which only required one API key (default builtin key or a user-specified key to avoid API rate limits). Check #541 for more details.
